CMakeLists.txt

```diff
@@ -208,48 +212,49 @@ endif()
-  if(result)
-      message(FATAL_ERROR "CMake step for googletest failed: ${result}")
-  endif()
-  execute_process(COMMAND ${CMAKE_COMMAND} --build .
-  RESULT_VARIABLE result
-  WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/gtest-download )
-  if(result)
-      message(FATAL_ERROR "Build step for googletest failed: ${result}")
-  endif()
-
-  # Prevent overriding the parent project's compiler/linker
-  # settings on Windows
-  set(gtest_force_shared_crt ON CACHE BOOL "" FORCE)
-
-  # Add googletest directly to our build. This defines
-  # the gtest and gtest_main targets.
-  add_subdirectory(${CMAKE_CURRENT_BINARY_DIR}/gtest-src
-                   ${CMAKE_CURRENT_BINARY_DIR}/gtest-build
-                   EXCLUDE_FROM_ALL)
-
-  # The gtest/gtest_main targets carry header search path
-  # dependencies automatically when using CMake 2.8.11 or
-  # later. Otherwise we have to add them here ourselves.
-  if (CMAKE_VERSION VERSION_LESS 2.8.11)
-      include_directories("${gtest_SOURCE_DIR}/include")
-      include_directories("${gmock_SOURCE_DIR}/include")
+if(ENABLE_GOOGLEMOCK)
+
+  include(FindGTest)
+  if (NOT GTest_FOUND AND GIT_FOUND) 
+    # Download and unpack googletest at configure time
+    configure_file(gtest/CMakeLists.txt.in gtest-download/CMakeLists.txt)
+    execute_process(COMMAND ${CMAKE_COMMAND} -G "${CMAKE_GENERATOR}" .
+    RESULT_VARIABLE result
+    WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/gtest-download )
+    if(result)
+        message(FATAL_ERROR "CMake step for googletest failed: ${result}")
+    endif()
+    execute_process(COMMAND ${CMAKE_COMMAND} --build .
+    RESULT_VARIABLE result
+    WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/gtest-download )
+    if(result)
+        message(FATAL_ERROR "Build step for googletest failed: ${result}")
+    endif()
+  
+    # Prevent overriding the parent project's compiler/linker
+    # settings on Windows
+    set(gtest_force_shared_crt ON CACHE BOOL "" FORCE)
+  
+    # Add googletest directly to our build. This defines
+    # the gtest and gtest_main targets.
+    add_subdirectory(${CMAKE_CURRENT_BINARY_DIR}/gtest-src
+                     ${CMAKE_CURRENT_BINARY_DIR}/gtest-build
+                     EXCLUDE_FROM_ALL)
+  
+    # The gtest/gtest_main targets carry header search path
+    # dependencies automatically when using CMake 2.8.11 or
+    # later. Otherwise we have to add them here ourselves.
+    if (CMAKE_VERSION VERSION_LESS 2.8.11)
+        include_directories("${gtest_SOURCE_DIR}/include")
+        include_directories("${gmock_SOURCE_DIR}/include")
+    endif()
   endif()
 
   add_subdirectory(tests)
-else()
- if (ENABLE_GOOGLEMOCK)
-    message(WARNING "googletest based unit tests disabled due to missing git. Please install git.")
- endif(ENABLE_GOOGLEMOCK)
 endif()
+if (ENABLE_GOOGLEMOCK AND NOT GTest_FOUND AND NOT GIT_FOUND)
+  message(WARNING "googletest based unit tests disabled due to missing googletest or git. Please install one of those.")
```

Comment:
@narc-Ontakac2:
this implies that if both gtest and git were missing, the tests would be skipped.
while this is very unlikely to happen in the "gitlab action" (as both are explicitly installed, and git is kind of required),
this should still be fatal instead of a warning in that case.
yes, it's somewhat esoteric.